compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -174,17 +178,19 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
           val mutable = true // static fields must always be mutable
           val name = js.StringLiteral(export.jsName)
           val irTpe = genExposedFieldIRType(fieldSym)
-          js.FieldDef(static = true, name, irTpe, mutable)
+          checkedCast[A](js.FieldDef(static = true, name, irTpe, mutable))
         } else {
-          js.TopLevelFieldExportDef(export.jsName, encodeFieldSym(fieldSym))
+          checkedCast[A](
+              js.TopLevelFieldExportDef(export.jsName, encodeFieldSym(fieldSym)))
         }
 
         (tree, export.jsName, pos)
       }).toList
     }
 
-    private def genTopLevelOrStaticMethodExports(classSym: Symbol,
-        destination: ExportDestination): List[(js.Tree, String, Position)] = {
+    private def genTopLevelOrStaticMethodExports[A <: js.IRNode: ClassTag](
+        classSym: Symbol,
+        destination: ExportDestination): List[(A, String, Position)] = {
```

Comment:
Just to follow up the conversation we had over the weekend: I think it is completely OK to "leak" the implementation detail of which tree we use for a given `ExportDestination`. IMO this is still better than having these strange cast incantations requiring `ClassTag` and what not.